Content.Shared/Slippery/SlidingSystem.cs

```diff
@@ -25,28 +48,64 @@ private void OnStand(EntityUid uid, SlidingComponent component, ref StoodEvent a
     }
 
     /// <summary>
-    ///     Sets friction to 0 if colliding with a SuperSlippery Entity.
+    ///     Updates friction when we collide with a slippery entity
     /// </summary>
-    private void OnStartCollide(EntityUid uid, SlidingComponent component, ref StartCollideEvent args)
+    private void OnStartCollide(Entity<SlidingComponent> entity, ref StartCollideEvent args)
     {
-        if (!TryComp<SlipperyComponent>(args.OtherEntity, out var slippery) || !slippery.SlipData.SuperSlippery)
+        if (!TryComp<SlipperyComponent>(args.OtherEntity, out var slippery) || !slippery.AffectsSliding)
             return;
 
-        component.CollidingEntities.Add(args.OtherEntity);
-        Dirty(uid, component);
+        if (!CalculateSlidingModifier(entity))
+            throw new Exception($"Entity by the name of {ToPrettyString(entity)} was given the Sliding Component despite not colliding with anything slippery");
+
+        _speedModifierSystem.RefreshFrictionModifiers(entity);
     }
 
     /// <summary>
-    ///     Set friction to normal when ending collision with a SuperSlippery entity.
+    ///     Update friction when we stop colliding with a slippery entity
     /// </summary>
-    private void OnEndCollide(EntityUid uid, SlidingComponent component, ref EndCollideEvent args)
+    private void OnEndCollide(Entity<SlidingComponent> entity, ref EndCollideEvent args)
     {
-        if (!component.CollidingEntities.Remove(args.OtherEntity))
+        if (!TryComp<SlipperyComponent>(args.OtherEntity, out var slippery) || !slippery.AffectsSliding)
             return;
 
-        if (component.CollidingEntities.Count == 0)
-            RemComp<SlidingComponent>(uid);
+        if (!CalculateSlidingModifier(entity, args.OtherEntity))
+        {
+            entity.Comp.FrictionModifier = 1;
+            RemComp<SlidingComponent>(entity);
+        }
+
+        _speedModifierSystem.RefreshFrictionModifiers(entity);
+    }
 
-        Dirty(uid, component);
+    /// <summary>
+    ///     Gets contacting slippery entities and averages their friction modifiers.
+    /// </summary>
+    private bool CalculateSlidingModifier(Entity<SlidingComponent> entity, EntityUid? ignore = null, PhysicsComponent? physics = null)
```

Comment:
```suggestion
    private bool CalculateSlidingModifier(Entity<SlidingComponent, PhysicsComponent?> entity, EntityUid? ignore = null)
```
do a resolve for PhysicsComponent, that way passing it in is optional